join.go

```diff
@@ -24,9 +24,22 @@ func IsNotExist(err error) bool {
 	return errors.Is(err, os.ErrNotExist) || errors.Is(err, syscall.ENOTDIR) || errors.Is(err, syscall.ENOENT)
 }
 
-// errUncleanRoot is returned if the user provides SecureJoinVFS with a path
-// that is not filepath.Clean'd.
-var errUncleanRoot = errors.New("root path provided to SecureJoin was not filepath.Clean")
+// errUnsafeRoot is returned if the user provides SecureJoinVFS with a path
+// that contains ".." components.
+var errUnsafeRoot = errors.New("root path provided to SecureJoin contains '..' components")
+
+// hasDotDot checks if the path contains ".." components in a platform-agnostic
+// way.
+func hasDotDot(path string) bool {
+	if v := filepath.VolumeName(path); v != "" {
+		path = path[len(v):]
+	}
```

Comment:
Can be simplified to
```go
        path = path[len(VolumeName(path)):]
```

The whole thing will probably be optimized out by the compiler on Linux since VolumeName always returns "" there. Or maybe I am being overly optimistic wrt Go compiler :)

Reply:
We do the same thing later for `remainingPath`, I'll combine them into one function.

Based on some [quick godbolt tests](https://godbolt.org/z/9e6fqEenf), the Go compiler is smart enough to make it a no-op but if you put it into a function the function is still included even though it's not used (though I guess the linker might get rid of it so looking at godbolt might not show that?).